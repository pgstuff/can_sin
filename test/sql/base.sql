\set ECHO none
\i sql/can_sin.sql
\set ECHO all

CREATE TABLE can_sins(id serial primary key, can_sin can_sin unique);
INSERT INTO can_sins(can_sin) VALUES('046 454 286');
INSERT INTO can_sins(can_sin) VALUES('000000000'); -- test limits
INSERT INTO can_sins(can_sin) VALUES('999 999 998');

SELECT * FROM can_sins ORDER BY can_sin;

SELECT MIN(can_sin) AS min FROM can_sins;
SELECT MAX(can_sin) AS max FROM can_sins;

SELECT id,can_sin::text FROM can_sins WHERE can_sin = '999 999 998';

-- index scan
SELECT id,can_sin::text FROM can_sins WHERE can_sin = '000000000';

SET enable_seqscan = false;
SELECT id,can_sin::text FROM can_sins WHERE can_sin = '999 999 998';
RESET enable_seqscan;

-- operators and conversions
SELECT '000000000'::can_sin < '000000000'::can_sin;
SELECT '000000000'::can_sin > '000000000'::can_sin;
SELECT '000000000'::can_sin < '999999998'::can_sin;
SELECT '000000000'::can_sin > '999999998'::can_sin;
SELECT '000000000'::can_sin <= '000000000'::can_sin;
SELECT '000000000'::can_sin >= '000000000'::can_sin;
SELECT '000000000'::can_sin <= '999999998'::can_sin;
SELECT '000000000'::can_sin >= '999999998'::can_sin;
SELECT '000000000'::can_sin <> '000000000'::can_sin;
SELECT '000000000'::can_sin <> '999999998'::can_sin;
SELECT '000000000'::can_sin = '000000000'::can_sin;
SELECT '000000000'::can_sin = '999999998'::can_sin;

-- COPY FROM/TO
TRUNCATE can_sins;
COPY can_sins(can_sin) FROM STDIN;
000000000
999 999 998
\.
COPY can_sins TO STDOUT;

-- clean up --
DROP TABLE can_sins;

-- errors
SELECT ''::can_sin;
SELECT '823 456 785'::can_sin; -- 8 is not used.
SELECT '1000000000'::can_sin;
SELECT '!'::can_sin;
