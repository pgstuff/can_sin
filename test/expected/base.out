\set ECHO none
CREATE TABLE can_sins(id serial primary key, can_sin can_sin unique);
INSERT INTO can_sins(can_sin) VALUES('046 454 286');
INSERT INTO can_sins(can_sin) VALUES('000000000'); -- test limits
INSERT INTO can_sins(can_sin) VALUES('999 999 998');
SELECT * FROM can_sins ORDER BY can_sin;
 id |   can_sin   
----+-------------
  2 | 000 000 000
  1 | 046 454 286
  3 | 999 999 998
(3 rows)

SELECT MIN(can_sin) AS min FROM can_sins;
     min     
-------------
 000 000 000
(1 row)

SELECT MAX(can_sin) AS max FROM can_sins;
     max     
-------------
 999 999 998
(1 row)

SELECT id,can_sin::text FROM can_sins WHERE can_sin = '999 999 998';
 id |   can_sin   
----+-------------
  3 | 999 999 998
(1 row)

-- index scan
SELECT id,can_sin::text FROM can_sins WHERE can_sin = '000000000';
 id |   can_sin   
----+-------------
  2 | 000 000 000
(1 row)

SET enable_seqscan = false;
SELECT id,can_sin::text FROM can_sins WHERE can_sin = '999 999 998';
 id |   can_sin   
----+-------------
  3 | 999 999 998
(1 row)

RESET enable_seqscan;
-- operators and conversions
SELECT '000000000'::can_sin < '000000000'::can_sin;
 ?column? 
----------
 f
(1 row)

SELECT '000000000'::can_sin > '000000000'::can_sin;
 ?column? 
----------
 f
(1 row)

SELECT '000000000'::can_sin < '999999998'::can_sin;
 ?column? 
----------
 t
(1 row)

SELECT '000000000'::can_sin > '999999998'::can_sin;
 ?column? 
----------
 f
(1 row)

SELECT '000000000'::can_sin <= '000000000'::can_sin;
 ?column? 
----------
 t
(1 row)

SELECT '000000000'::can_sin >= '000000000'::can_sin;
 ?column? 
----------
 t
(1 row)

SELECT '000000000'::can_sin <= '999999998'::can_sin;
 ?column? 
----------
 t
(1 row)

SELECT '000000000'::can_sin >= '999999998'::can_sin;
 ?column? 
----------
 f
(1 row)

SELECT '000000000'::can_sin <> '000000000'::can_sin;
 ?column? 
----------
 f
(1 row)

SELECT '000000000'::can_sin <> '999999998'::can_sin;
 ?column? 
----------
 t
(1 row)

SELECT '000000000'::can_sin = '000000000'::can_sin;
 ?column? 
----------
 t
(1 row)

SELECT '000000000'::can_sin = '999999998'::can_sin;
 ?column? 
----------
 f
(1 row)

-- COPY FROM/TO
TRUNCATE can_sins;
COPY can_sins(can_sin) FROM STDIN;
COPY can_sins TO STDOUT;
4	000 000 000
5	999 999 998
-- clean up --
DROP TABLE can_sins;
-- errors
SELECT ''::can_sin;
ERROR:  Invalid number of digits (0) in input data "".
LINE 1: SELECT ''::can_sin;
               ^
HINT:  Valid CANSIN consists of 9 digits.
SELECT '823 456 785'::can_sin; -- 8 is not used.
ERROR:  CANSIN number 823 456 785 is invalid.
LINE 1: SELECT '823 456 785'::can_sin;
               ^
SELECT '1000000000'::can_sin;
ERROR:  CANSIN number "1000000000" is invalid.
LINE 1: SELECT '1000000000'::can_sin;
               ^
SELECT '!'::can_sin;
ERROR:  Unexpected character '!' in input data "!".
LINE 1: SELECT '!'::can_sin;
               ^
HINT:  Valid CANSIN consists of digits and optional spaces.
